test/e2e/collect-logs.sh

```diff
@@ -0,0 +1,169 @@
+    fi
+    
+    # Collect logs for each container
+    for container in $containers; do
+        log_file="${log_file_prefix}-${container}.log"
+        echo "  - Container ${container} -> ${log_file}"
+        
+        # Get current logs
+        kubectl logs "${pod_name}" -n "${NAMESPACE}" -c "${container}" > "${log_file}" 2>&1 || \
+            echo "Failed to get logs for container ${container}" > "${log_file}"
+        
+        # Try to get previous logs if pod was restarted
+        previous_log_file="${log_file_prefix}-${container}-previous.log"
+        if kubectl logs "${pod_name}" -n "${NAMESPACE}" -c "${container}" --previous > "${previous_log_file}" 2>&1; then
+            echo "  - Previous logs for ${container} -> ${previous_log_file}"
+        else
+            rm -f "${previous_log_file}"
+        fi
+    done
+}
+
+# Collect hub cluster logs
+echo -e "${GREEN}=== Collecting Hub Cluster Logs ===${NC}"
+kind export kubeconfig --name "${HUB_CLUSTER}" 2>/dev/null || {
+    echo -e "${RED}Failed to export kubeconfig for hub cluster${NC}"
+    exit 1
+}
+
+# Create hub logs directory
+HUB_LOG_DIR="${LOG_DIR}/hub"
+mkdir -p "${HUB_LOG_DIR}"
+
+# Get all hub-agent pods
+hub_pods=$(kubectl get pods -n "${NAMESPACE}" -l app.kubernetes.io/name=hub-agent -o jsonpath='{.items[*].metadata.name}' 2>/dev/null || echo "")
+
+if [ -z "$hub_pods" ]; then
+    echo -e "${RED}No hub-agent pods found${NC}"
+else
+    for pod in $hub_pods; do
+        collect_pod_logs "${pod}" "${HUB_CLUSTER}" "${HUB_LOG_DIR}/${pod}"
+    done
+fi
+
+# Collect member cluster logs
+for cluster in "${MEMBER_CLUSTERS[@]}"; do
+    echo -e "${GREEN}=== Collecting Member Cluster Logs: ${cluster} ===${NC}"
+    
+    # Export kubeconfig for the member cluster
+    if ! kind export kubeconfig --name "${cluster}" 2>/dev/null; then
+        echo -e "${RED}Failed to export kubeconfig for cluster ${cluster}, skipping...${NC}"
+        continue
+    fi
+    
+    # Create member logs directory
+    MEMBER_LOG_DIR="${LOG_DIR}/${cluster}"
+    mkdir -p "${MEMBER_LOG_DIR}"
+    
+    # Get all member-agent pods
+    member_pods=$(kubectl get pods -n "${NAMESPACE}" -l app.kubernetes.io/name=member-agent -o jsonpath='{.items[*].metadata.name}' 2>/dev/null || echo "")
+    
+    if [ -z "$member_pods" ]; then
+        echo -e "${RED}No member-agent pods found in cluster ${cluster}${NC}"
+    else
+        for pod in $member_pods; do
+            collect_pod_logs "${pod}" "${cluster}" "${MEMBER_LOG_DIR}/${pod}"
+        done
+    fi
+    
+    echo ""
+done
+
+# Collect additional debugging information
+echo -e "${GREEN}=== Collecting Additional Debug Information ===${NC}"
+
+# Hub cluster debug info
+kind export kubeconfig --name "${HUB_CLUSTER}" 2>/dev/null
+{
+    echo "=== Hub Cluster Pod Status ==="
+    kubectl get pods -n "${NAMESPACE}" -o wide
+    echo ""
+    echo "=== Hub Cluster Events ==="
+    kubectl get events -n "${NAMESPACE}" --sort-by='.lastTimestamp'
```

Comment:
The events are collected in the fleet-system namespace. Will we get anything real?